README.md

```diff
@@ -22,22 +22,107 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+Add an instance of GraphQL::Analyzer to your schema, instantiate it with a list of instrumentation to capture different datastore queries.
+
+### Analyzer
+
+Add 'GraphQL::Analyzer' to your schema:
+
+```ruby
+require 'graphql/analyzer'
+
+Schema = GraphQL::Schema.define do
+  query QueryType
+  use(
+      GraphQL::Analyzer.new(
+        GraphQL::Analyzer::Instrumentation::Mysql.new,
+        GraphQL::Analyzer::Instrumentation::Postgresql.new
+      )
+  )
+end
+```
+
+### Instrumentation
+
+To write your own custom instrumentation, your object needs to respond to `#instrument(type, field)` and return a lambda that accepts three parameters, `object`, `arguments`, and `context`, and returns the original field value. It should also add any queries captured to the `context`.
+
+```ruby
+module GraphQL
+  class Analyzer
+    module Instrumentation
+      class MyCustomInstrumentation < Base
+        def instrument(type, field)
+          ->(obj, args, ctx) do
+            ### OMITTED ###
+              ctx['graphql-analyzer']['resolvers'] << {
+                'path' => ctx.path,
+                'adapter' => adapter,
+                'parentType' => type.name,
+                'fieldName' => field.name,
+                'returnType' => field.type.to_s,
+                'details' => parsed_output
+              }
+              ### OMITTED ###
+          end
+        end
+      end
+    end
+  end
+end
+```
+
+### Response Format
```

Comment:
Given that the instrumentation section is optional, what do you think of putting it below the Response Format section?

Reply:
👍 I'll also add the list of already implemented instruments to this section.